Source/NetUtilities.swift

```diff
@@ -0,0 +1,111 @@
+//  Copyright © 2016 Venture Media Labs. All rights reserved.
+//
+// This file is part of BrainCore. The full BrainCore copyright notice,
+// including terms governing use, modification, and redistribution, is
+// contained in the file LICENSE at the root of the source code distribution
+// tree.
+
+import Metal
+
+
+
+infix operator => { associativity left precedence 160 }
+public func =>(lhs: Layer, rhs: Layer) -> Layer {
+    let net = Net.buildNet
+
+    let lhsLayer: Net.LayerRef
+    if let node = net.nodeWithLayerName(lhs.name) {
+        lhsLayer = node.id
+    } else {
+        lhsLayer = net.addLayer(lhs)
+    }
+
+    let rhsLayer: Net.LayerRef
+    if let node = net.nodeWithLayerName(rhs.name) {
+        rhsLayer = node.id
+    } else {
+        rhsLayer = net.addLayer(rhs)
+    }
+
+    let buff = net.addBufferWithName("\(lhs.name) -> \(rhs.name)")
+
+    net.connectLayer(lhsLayer, toBuffer: buff)
+    net.connectBuffer(buff, toLayer: rhsLayer)
+
+    return rhs
+}
+
+public func =>(lhs: [Layer], rhs: Layer) -> Layer {
+    let net = Net.buildNet
+
+    var leftLayers = Array<Net.LayerRef>()
+    for layer in lhs {
+        if let node = net.nodeWithLayerName(layer.name) {
+            leftLayers.append(node.id)
+        } else {
+            leftLayers.append(net.addLayer(layer))
+        }
+    }
+
+    var rightLayer: Net.LayerRef
+    if let node = net.nodeWithLayerName(rhs.name) {
+        rightLayer = node.id
+    } else {
+        rightLayer = net.addLayer(rhs)
+    }
+
+    let buff = net.addBufferWithName("\(lhs.map({ $0.name })) -> \(rhs.name)")
+
+    var offset = 0
+    for layerRef in leftLayers {
+        var outputSize: Int
+        if let dataLayer = net.nodes[layerRef].layer as? DataLayer {
+            outputSize = dataLayer.outputSize
+        } else if let forwardLayer = net.nodes[layerRef].layer as? ForwardLayer {
+            outputSize = forwardLayer.outputSize
+        } else {
+            fatalError("Cannot connect \(net.nodes[layerRef].layer.dynamicType)")
+        }
+
+        net.connectLayer(layerRef, toBuffer: buff, atOffset: offset)
+        offset += outputSize
+    }
+    net.connectBuffer(buff, toLayer: rightLayer)
+
+    return rhs
+}
+
+public func =>(lhs: (bufferRef: Net.BufferRef, offset: Int), rhs: Layer) -> Layer {
+    let net = Net.buildNet
+
+    var rhsRef: Net.LayerRef
+    if let node = net.nodeWithLayerName(rhs.name) {
+        rhsRef = node.id
+    } else {
+        rhsRef = net.addLayer(rhs)
+    }
+
+    net.connectBuffer(lhs.bufferRef, atOffset: lhs.offset, toLayer: rhsRef)
+
+    return rhs
+}
+
+public extension Layer {
+    public subscript(offset: Int) -> (bufferRef: Net.BufferRef, offset: Int) {
```

Comment:
Also add a documentation block explaining what this does and why it doesn't take a size.